src/Ensconce.Cli/ApplicationInteraction.cs

```diff
@@ -15,66 +16,100 @@ internal static class ApplicationInteraction
-                    service.Stop();
-                }
+                    var service = ServiceController.GetServices().First(svc => svc.DisplayName == serviceDetails.DisplayName);
+                    if (service.Status != ServiceControllerStatus.Stopped)
+                    {
+                        Logging.Log("Stopping service {0}", serviceDetails.DisplayName);
+                        service.Stop();
+                    }
 
-                VerifyServiceStopped(serviceDetails.DisplayName, 30);
+                    VerifyServiceStopped(serviceDetails.DisplayName, 30);
 
-                Logging.Log("Uninstalling service {0}", serviceDetails.DisplayName);
-                Process.Start("sc", string.Format("delete \"{0}\"", serviceDetails.ServiceName)).WaitForExit();
+                    Logging.Log("Uninstalling service {0}", serviceDetails.DisplayName);
+                    Process.Start("sc", $"delete \"{serviceDetails.ServiceName}\"")?.WaitForExit();
 
-                VerifyServiceUninstall(serviceDetails.DisplayName, 30);
+                    VerifyServiceUninstall(serviceDetails.DisplayName, 30);
+                }
+            }
+            else
+            {
+                Logging.Log("No services found in {0}", directory);
             }
         }
 
         internal static void StopProcessesInDirectory(string directory)
         {
-            if (!directory.EndsWith(@"\"))
+            Logging.Log("Stopping processes in directory: {0}", directory);
+
+            var processes = GetProcessesRunningInDirectory(directory);
+            if (!directory.EndsWith(Path.DirectorySeparatorChar.ToString()))
             {
-                directory = directory + @"\";
+                directory += Path.DirectorySeparatorChar;
             }
 
-            Logging.Log("Stopping processes in directory: {0}", directory);
-            var wmiQueryString = "SELECT ProcessId, ExecutablePath, CommandLine FROM Win32_Process";
+            if (processes.Any())
+            {
+                Logging.Log("{0} processes found in {1}", processes.Count, directory);
 
-            using (var searcher = new ManagementObjectSearcher(wmiQueryString))
-            using (var results = searcher.Get())
+                foreach (var process in processes)
+                {
+                    try
+                    {
+                        Logging.Log("Attempting to kill {0}", process.Path);
+                        process.Process.Kill();
+                    }
+                    catch (Win32Exception ex) when (ex.NativeErrorCode == 5) //Access Denied
+                    {
+                        // Process is already terminating.
+                    }
+                    catch (InvalidOperationException)
+                    {
+                        // Process has already terminated.
+                    }
+
+                    process.Process.WaitForExit((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
```

Comment:
`(int)TimeSpan.FromSeconds(30).TotalMilliseconds`

Complex way of writing 30_000 😆